src/lib/core/ripple/ripple-renderer.ts

```diff
@@ -143,6 +143,8 @@ export class RippleRenderer {
 
     rippleDiv.addEventListener('transitionend',
         (event: TransitionEvent) => transitionEndCallback(ripple, event));
+    // Remove the ripple after transitions (fade-in, transform, fade-out)
+    setTimeout(() => this.removeRippleFromDom(ripple.rippleElement), fadeInSeconds * 3 * 1000);
```

Comment:
```ts
// Ensure that ripples are always removed, even when transitionend doesn't fire.
// Run this outside the Angular zone because there's nothing that Angular cares about.
// If it were to run inside the Angular zone, every test that used ripples would have to be
// either async or fakeAsync.
this._ngZone.runOutsideAngular(() => {
  // The ripple lasts a time equal to the sum of fade-in, transform, and fade-out (3 * fade-in time).
  let rippleDuration =  fadeInSeconds * 3 * 1000;
  setTimeout(() => this.removeRippleFromDom(ripple.rippleElement), rippleDuration);
});